broker/src/main/java/io/zeebe/broker/system/partitions/impl/AsyncSnapshotDirector.java

```diff
@@ -108,6 +116,40 @@ public void forceSnapshot() {
     actor.call(this::prepareTakingSnapshot);
   }
 
+  @Override
+  protected void handleFailure(final Exception failure) {
+    LOG.error(
+        "Snapshot director '{}' failed in phase {} with: {}",
+        actorName,
+        actor.getLifecyclePhase(),
+        failure);
+    actor.fail();
```

Comment:
I'm wondering if we should hard fail here. It is possible to recover when it takes the next snapshot, right? If the corruption is in one segment, but the processor and exporter have already moved past that segment, we would be able to take the next snapshot and compact the corrupted segment.

Reply:
I guess it depends on whether the exporter and snapshot have passed that position or not. We can set the snapshot director's health to unhealthy and back to healthy when it successfully takes a snapshot

Reply:
This was more of a question than request to change. It would be good to discuss and decide what would be a good behavior. If it goes only unhealthy, then we also need to ensure that it's state is rolled back - pending snapshot is aborted, all other internal actor states are reverted so that next snapshot can be successfully taken (if possible).